src/transformers/modeling_tf_utils.py

```diff
@@ -215,77 +218,184 @@ class TFMaskedLanguageModelingLoss(TFCausalLanguageModelingLoss):
+    .. note::
+
+         Any label of -100 will be ignored (along with the corresponding logits) in the loss computation.
+    """
+
+    def compute_loss(self, labels, logits):
+        loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(
+            from_logits=True, reduction=tf.keras.losses.Reduction.NONE
+        )
+        # make sure only labels that are not equal to -100
+        # are taken into account as loss
+        next_sentence_active_loss = tf.not_equal(tf.reshape(labels, (-1,)), -100)
+        next_sentence_reduced_logits = tf.boolean_mask(tf.reshape(logits, (-1, 2)), next_sentence_active_loss)
+        next_sentence_label = tf.boolean_mask(tf.reshape(labels, (-1,)), next_sentence_active_loss)
+
+        return loss_fn(next_sentence_label, next_sentence_reduced_logits)
+
+
+def input_analysis(func, inputs, **kwargs):
+    signature = inspect.signature(func).parameters
+    parameter_names = list(signature.keys())
+    output = {}
+
+    if isinstance(inputs, (tuple, list)):
+        for i, input in enumerate(inputs):
+            if type(input) == tf.Tensor:
+                if input.name in parameter_names:
+                    output[input.name] = input
+                else:
+                    raise ValueError(
+                        "The tensor named %s does not belong to the authorized list of names %s "
+                        % (input.name, parameter_names)
+                    )
+            elif isinstance(input, tf.Tensor) or input is None:
+                output[parameter_names[i]] = input
+            else:
+                raise ValueError("Data of type %s is not allowed only tf.Tensor is accepted." % type(input))
+    elif isinstance(inputs, (dict, BatchEncoding)):
+        output = dict(inputs)
+
+        for k, v in output.items():
+            if not isinstance(v, tf.Tensor):
+                raise ValueError("Data of type %s is not allowed only tf.Tensor is accepted." % type(input))
+    else:
+        if isinstance(inputs, tf.Tensor) or inputs is None:
+            output[parameter_names[0]] = inputs
+        else:
+            raise ValueError("Data of type %s is not allowed only tf.Tensor is accepted." % type(inputs))
+
+    for name in parameter_names:
+        if name not in list(output.keys()):
+            # When creating a SavedModel TF calls method the method LayerCall.__call__(args, **kwargs)
+            # So to respect the proper output we have to add this exception
+            if name == "kwargs":
+                output.update(kwargs)
+
+                if "input_ids" not in list(output.keys()):
+                    output["input_ids"] = output["args"]
+                elif "input_embeds" not in list(output.keys()):
+                    output["input_embeds"] = output["args"]
+
+                del output["args"]
+            else:
+                output[name] = kwargs.pop(name, signature[name].default)
+
+    return output
+
+
+def old_load_tf_weights(model, resolved_archive_file):
```

Comment:
Do we need to keep this function?

Reply:
Until all the models will adopt the same changes than what I did in BERT.